components/prism-core.js

```diff
@@ -447,15 +589,50 @@ var _ = {
 
 	Token: Token
 };
-
 _self.Prism = _;
 
+/**
+ * Creates a new token.
+ *
+ * @param {string} type See {@link Prism.Token#type type}
+ * @param {string | Prism.Token | Array.<string|Prism.Token>} content See {@link Prism.Token#content content}
+ * @param {string|string[]} [alias] The alias(es) of the token.
+ * @param {string} [matchedStr=""] A copy of the full string this token was created from.
+ * @param {boolean} [greedy=false] See {@link Prism.Token#greedy greedy}
+ * @class
+ * @memberof Prism
+ */
 function Token(type, content, alias, matchedStr, greedy) {
+	/**
+	 * The type of the token.
+	 *
+	 * This is usually the key of a pattern in a {@link Grammar}.
+	 * @member {string}
+	 */
 	this.type = type;
+	/**
+	 * The strings or tokens contained by this token.
+	 *
+	 * This will be a token stream if the pattern matched also defined an `inside` grammar.
+	 * @member {string | Prism.Token | Array.<string|Prism.Token>}
+	 */
 	this.content = content;
+	/**
+	 * The alias(es) of the token.
+	 * @member {string|string[]}
+	 * @see TokenObject
+	 */
 	this.alias = alias;
-	// Copy of the full string this token was created from
+	/**
+	 *
+	 * @member {number}
+	 */
 	this.length = (matchedStr || "").length|0;
+	/**
+	 * Whether the pattern that created this token is greedy or not.
+	 * @member {boolean}
+	 * @see TokenObject
+	 */
 	this.greedy = !!greedy;
 }
 
```

Comment:
This adds JSDoc for&nbsp;the&nbsp;`Token.stringify(…)` static method.
```suggestion

/**
 * Converts the token to an HTML representation.
 *
 * @param {string|Token|Token[]} o The token to convert to an HTML representation.
 * @param {string} language The current programming language.
 * @param {string|Token|Token[]} [parent] The parent token, if any.
 * @return {string} The HTML representation of the token.
 */
```

Reply:
I added docs for it but made them private because I don't think that this method, which only ever used by prism-core itself, should appear in the _public_ API doc.

Reply:
It’s still good to have that for type checking.

Reply:
Well, it still isn't working properly because of the `@memberof`s...